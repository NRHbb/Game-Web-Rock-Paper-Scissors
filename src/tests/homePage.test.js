/** @jest-environment jsdom */
import React from 'react'
import { render } from './testUtils.js'
import Home from '../pages/home/index.jsx'
import { useRouter } from 'next/router'

jest.mock('next/router', () => ({
  useRouter: jest.fn()
}))

const mockRouter = {
  route: '/mock-route',
  basePath: '',
  pathname: '/mock-route',
  query: {},
  asPath: '/mock-route'
}

useRouter.mockReturnValue(mockRouter)

describe('render test', () => {
  it('renders home page without crashing', () => {
    const home = render(<Home />)
    expect(home).toMatchSnapshot()
  })
})
