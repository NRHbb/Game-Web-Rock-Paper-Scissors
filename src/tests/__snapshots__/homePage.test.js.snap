// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test renders home page without crashing 1`] = `
<Provider
  store={
    {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
